russstation/code/controllers/subsystem/vote.dm

```diff
@@ -1,19 +1,30 @@
-/datum/controller/subsystem/vote/
-	var/shuttle_refuel_delay = 0
-	var/transfer_vote_config = 0
+/datum/controller/subsystem/vote/var/transit_timer
 
-/datum/controller/subsystem/vote/proc/shuttlecall()
-	var/shuttle_timer = SSshuttle.emergency.timeLeft()
-	SSshuttle.block_recall(6000)
-	if(shuttle_timer >= 6000 || (SSshuttle.emergency.mode != SHUTTLE_CALL && SSshuttle.emergency.mode != SHUTTLE_DOCKED && SSshuttle.emergency.mode != SHUTTLE_ESCAPE))
-		if(SSshuttle.emergency.mode == SHUTTLE_CALL && shuttle_timer >= 6000)	//Apparently doing the emergency request twice cancels the call so these check are just in case
-			SSshuttle.emergency.setTimer(6000)
-			priority_announce("The emergency shuttle will arrive in [SSshuttle.emergency.timeLeft()/60] minutes.")
-		else if (SSshuttle.emergency.mode != SHUTTLE_CALL)
-			SSshuttle.emergency.request()
-			SSshuttle.emergency.setTimer(6000)
-			priority_announce("The emergency shuttle will arrive in [SSshuttle.emergency.timeLeft()/60] minutes.")
+/datum/controller/subsystem/vote/OnConfigLoad()
+	if(CONFIG_GET(flag/transfer_vote))
+		var/initial_delay = CONFIG_GET(number/transfer_delay_initial)
+		if(initial_delay > world.time)
+			set_transfer_timer(initial_delay - world.time)
+		else
+			set_transfer_timer(CONFIG_GET(number/transfer_delay_subsequent))
+
+/datum/controller/subsystem/vote/proc/set_transfer_timer(timing)
+	if(transit_timer)
+		deltimer(transit_timer)
+	transit_timer = addtimer(CALLBACK(src, .proc/attempt_transfer_vote), timing)
 
-		message_admins("The emergency shuttle has been force-called due to a successful crew transfer vote.")
+/datum/controller/subsystem/vote/proc/attempt_transfer_vote()
+	if(SSshuttle.emergency.mode == SHUTTLE_IDLE || SSshuttle.emergency.mode == SHUTTLE_RECALL)
+		initiate_vote("crew transfer","the server")
+	else
+		// recheck to see if the shuttle is no longer busy after ten minutes
+		set_transfer_timer(10 MINUTES)
+
+/datum/controller/subsystem/vote/proc/shuttlecall()
+	SSshuttle.emergency_no_recall = TRUE
```

Comment:
Should this check be set down below only upon successful shuttle recall? Currently it looks to be set even on failed votes.

Reply:
Oop, you're correct, totally forgot to move this while I was shifting things around. That being said this does present a difficulty. the `request()` call won't call the shuttle if it's in any state but idle, called, or recalled. We'd only want to block recall if it actually calls the shuttle, but `request()` doesn't return any value to indicate its success or not.

Reply:
Reconsidering that last comment, even if the call doesn't go through, that means the shuttle is current in some form of transit and therefore can be assumed as safe to block recall.

Reply:
There were a few states that didn't fit into the above category